crates/next-core/src/next_client_component/server_to_client_transition.rs

```diff
@@ -14,16 +25,44 @@ pub struct NextServerToClientTransition {
 #[turbo_tasks::value_impl]
 impl Transition for NextServerToClientTransition {
     #[turbo_tasks::function]
-    fn process_source(&self, asset: AssetVc) -> AssetVc {
-        VirtualAssetVc::new(
-            asset.ident().path().join("client-proxy.tsx"),
-            next_js_file(if self.ssr {
-                "entry/app/server-to-client-ssr.tsx"
-            } else {
-                "entry/app/server-to-client.tsx"
-            })
-            .into(),
+    async fn process(
+        self_vc: NextServerToClientTransitionVc,
+        asset: AssetVc,
+        context: ModuleAssetContextVc,
+        _reference_type: Value<ReferenceType>,
+    ) -> Result<AssetVc> {
+        let internal_asset = next_asset(if self_vc.await?.ssr {
+            "entry/app/server-to-client-ssr.tsx"
+        } else {
+            "entry/app/server-to-client.tsx"
+        });
+        let context = self_vc.process_context(context);
+        let client_chunks = context.with_transition("next-client-chunks").process(
+            asset,
+            Value::new(ReferenceType::Entry(
+                EntryReferenceSubType::AppClientComponent,
+            )),
+        );
+        let client_module = context.with_transition("next-ssr-client-module").process(
+            asset,
+            Value::new(ReferenceType::Entry(
+                EntryReferenceSubType::AppClientComponent,
+            )),
+        );
```

Comment:
It seems weird to put these transitions inside of the context, only to always apply them here. These `process` calls should just be methods on `NextServerToClientTransition`: e.g. `process_client_chunks`, `process_client_module`